src/geos.c

```diff
@@ -942,3 +942,37 @@ GEOSCoordSequence* coordseq_from_buffer(GEOSContextHandle_t ctx, const double* b
   }
   return coord_seq;
 }
+
+/* Copy coordinates of a GEOSCoordSequence to an array
+ *
+ * Note: this function assumes that the buffer is from a C-contiguous array,
+ * and that the dimension of the buffer is only 2D or 3D.
+ *
+ * Returns 0 on error, 1 on success.
+ */
+char coordseq_to_buffer(GEOSContextHandle_t ctx, const GEOSCoordSequence* coord_seq,
+                        double* buf, unsigned int size, unsigned int dims) {
+  char *cp1, *cp2;
+  unsigned int i, j;
+
+#if GEOS_SINCE_3_10_0
+
+  int hasZ = dims == 3;
+  if (!GEOSCoordSeq_copyToBuffer_r(ctx, coord_seq, buf, hasZ, 0)) {
```

Comment:
What about changing the return value to `int` and then 

```C
return GEOSCoordSeq_copyToBuffer_r(ctx, coord_seq, buf, hasZ, 0);
```

Reply:
Changed to int (I used char because some of our other helper functions were using that, in general we don't seem to be very consistent about using `int` vs `char` as return type)